File: PythonCodes/Fig01_01.py
#%%
"""
Created on Thu Nov 28 2018
Characteristic function and density for normal(10,1)
@author: Lech A. Grzelak
"""
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as st
from mpl_toolkits import mplot3d

def plotNormalPDF_CDF_CHF():
    mu    = 10.0
    sigma = 1.0
    i     = np.complex(0,1)
    chf   = lambda u: np.exp(i * mu * u - sigma * sigma * u * u / 2.0)
    pdf   = lambda x: st.norm.pdf(x,mu,sigma)
    cdf   = lambda x: st.norm.cdf(x,mu,sigma)

    x = np.linspace(5,15,100)
    u = np.linspace(0,5,250)
    
    # Figure 1, PDF

    plt.figure(1)
    plt.plot(x,pdf(x))
    plt.grid()
    plt.xlabel('x')
    plt.ylabel('PDF')
                     
    # Figure 2, CDF

    plt.figure(2)
    plt.plot(x,cdf(x))
    plt.grid()
    plt.xlabel('x')
    plt.ylabel('CDF')     
    
    # Figure 3, CHF

    plt.figure(3)
    ax = plt.axes(projection='3d')
    chfV = chf(u)
    
    x = np.real(chfV)
    y = np.imag(chfV)
    ax.plot3D(u, x, y, 'blue')
    ax.view_init(30, -120)
    
plotNormalPDF_CDF_CHF()

File: MatlabCodes/Fig01_01.m
function ChF_3D
close all;clc
mu    = 10;
sigma = 1;
i     = complex(0,1);
chf  = @(u)exp(i * mu * u - sigma * sigma * u .* u / 2.0);
pdf  = @(x)normpdf(x,mu,sigma);
cdf  = @(x)normcdf(x,mu,sigma);

% Density 

figure(1)
x = linspace(5,15,100);
plot(x,pdf(x),'linewidth',2)
grid on
xlabel('x')
ylabel('PDF')

% Cumulative distribution function

figure(2)
plot(x,cdf(x),'linewidth',2)
grid on
xlabel('x')
ylabel('CDF')

% 3D graph for the ChF

figure(3)
u = linspace(0,5,250);
chfVal = chf(u);
plot3(u,real(chfVal),imag(chfVal),'linewidth',2)
grid on
xlabel('u')
ylabel('real(ChF)')
zlabel('imag(ChF)')
